Teku Maven artifacts issues when including as dependency

<!-- 
By filing an Issue, you are expected to comply with the Code of Conduct, 
including treating everyone with respect:
https://github.com/Consensys/teku/blob/master/CODE-OF-CONDUCT.md

Not all sections will apply to all issue types.
-->

### Description

When using different versions of `io.spring.dependency-management` gradle plugin including any Teku module artifact as a maven dependency to a gradle build the following error may occur:

- version `1.1.0`: no errors
- version `1.1.1`: build fails with the following error (plugin issue [here](https://github.com/spring-gradle-plugins/dependency-management-plugin/issues/365))
```
> Task :startScripts FAILED
2 actionable tasks: 1 executed, 1 up-to-date
Errors occurred while build effective model from C:\Users\Admin\.gradle\caches\modules-2\files-2.1\tech.pegasys.teku.internal\unsigned\23.8.0\c0e66a134880efb10f2882a13b6cfd8d69b2fe7f\unsigned-23.8.0.pom:

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':startScripts'.
> Error while evaluating property 'relativeClasspath' of task ':startScripts'
   > Cannot invoke "io.spring.gradle.dependencymanagement.org.apache.maven.model.Model.getGroupId()" because "effectiveModel" is null
```
- version `1.1.3`: build succeeds bu the the following error(s) are printed:
```
Errors occurred while building effective model from C:\Users\Admin\.gradle\caches\modules-2\files-2.1\tech.pegasys.teku.internal\unsigned\23.8.0\c0e66a134880efb10f2882a13b6cfd8d69b2fe7f\unsigned-23.8.0.pom:
	'dependencies.dependency[tech.pegasys.teku.internal:unsigned:23.8.0]' for tech.pegasys.teku.internal:unsigned:23.8.0 is referencing itself. in tech.pegasys.teku.internal:unsigned:23.8.0
```

The above errors indicate another problem: all dependencies of `test-fixture` scope are also picked up when only module runtime dependencies are required. Here is some more insights: https://stackoverflow.com/questions/69877418/why-does-every-pom-file-published-by-gradle-has-a-self-referential-dependency



### Steps to Reproduce (Bug)
run `gradle build` for the following `build.gradle`

```gradle
plugins {
    id 'io.spring.dependency-management' version '1.1.1'
    id "java"
    id "application"
}

repositories {
        mavenCentral()
        maven { url "https://artifacts.consensys.net/public/teku/maven/" }
}

dependencies {
    implementation 'tech.pegasys.teku.internal:unsigned:23.8.0'
}
```

**Expected behavior:** 

Successful build
Only required artifacts are imported

**Actual behavior:** 

Build fails with the errors described above (depending on the plugin version)

**Frequency:** Always

### Versions (Add all that apply)
* Software version: 23.8.0
* Java version: 17
* OS Name & Version: Windows

### Related issues:
Libp2p PR: https://github.com/libp2p/jvm-libp2p/pull/306

--- a/build.gradle
+++ b/build.gradle
@@ -150,6 +150,9 @@ allprojects {
     }
   }
 
+  components.java.withVariantsFromConfiguration(configurations.testFixturesApiElements) { skip() }
+  components.java.withVariantsFromConfiguration(configurations.testFixturesRuntimeElements) { skip() }
+
   dependencies {
     errorprone("com.google.errorprone:error_prone_core")
     errorprone("tech.pegasys.tools.epchecks:errorprone-checks")

